sk_sidekick/lib/src/commands/release_command.dart

```diff
@@ -0,0 +1,315 @@
+      sleep(1);
+    }
+
+    while (!confirm(
+      'Do you want to continue the release process with the changelog in NEXT_RELEASE_CHANGELOG.md?',
+      defaultValue: false,
+    )) {}
+
+    final nextChangelogContent = nextReleaseChangelog.readAsStringSync();
+    nextReleaseChangelog.deleteSync();
+    return nextChangelogContent;
+  }
+
+  String? _getMinSidekickCoreVersion(String pubspec) {
+    final doc = loadYamlDocument(pubspec);
+    final ps = doc.contents.value as YamlMap;
+    final constraint = ps['dependencies']['sidekick_core'] as String?;
+    if (constraint == null) {
+      return null;
+    }
+    return VersionConstraint.parse(constraint).minVersion.toString();
+  }
+
+  List<String>? _getSidekickCoreChanges(DartPackage package) {
+    final currentSidekickCoreVersion =
+        _getMinSidekickCoreVersion(package.pubspec.readAsStringSync());
+    if (currentSidekickCoreVersion != null) {
+      final currentSidekickCoreVersionTag =
+          'sidekick_core-v$currentSidekickCoreVersion';
+
+      final oldPubspecContent =
+          'git show $currentSidekickCoreVersionTag:${package.name}/pubspec.yaml'
+              .start(progress: Progress.capture())
+              .lines
+              .join('\n');
+      final oldSidekickCoreVersion =
+          _getMinSidekickCoreVersion(oldPubspecContent);
+
+      if (oldSidekickCoreVersion != null &&
+          currentSidekickCoreVersion != oldSidekickCoreVersion) {
+        final oldSidekickCoreVersionTag =
+            'sidekick_core-v$oldSidekickCoreVersion';
+        final newSidekickCoreCommits = _getChanges(
+          from: oldSidekickCoreVersionTag,
+          to: currentSidekickCoreVersionTag,
+          paths: ['sidekick_core'],
+        );
+
+        final message = '\nAlso package:sidekick_core updated '
+            '($oldSidekickCoreVersionTag -> $currentSidekickCoreVersionTag), '
+            'please consider those changes as well.\n';
+
+        return [message, ...newSidekickCoreCommits];
+      }
+    }
+    return null;
+  }
+}
+
+/// Whether the git repository has untracked changes in [directory]
+///
+/// Note that the repository may contain untracked changes in other directories
+bool _gitRepoHasChangesIn(Directory directory) =>
+    'git status --porcelain ${directory.path}'
+        .start(progress: Progress.capture(), workingDirectory: directory.path)
+        .lines
+        .isNotEmpty;
+
+/// Returns all changes in the history from [from] (default: HEAD) to [to]
+/// If [paths] is given, returns only commits which modified [paths]
+///
+/// The format of each line is '<commit hash> <commit title>'
+List<String> _getChanges({
+  required String from,
+  String? to = 'HEAD',
+  List<String> paths = const [],
+}) =>
+    // %H = commit hash, %b = commit title
+    "git log --format='%H %s' $from..$to -- ${paths.join(' ')}"
+        .start(progress: Progress.capture())
+        .lines;
```

Comment:
Additionally, we should replace the PR link `(#123)` with a markdown link `[(#123)](https://github.com/phntmxyz/sidekick/pull/123)`

Reply:
I think the short commit hash is enough

Reply:
Good idea 👍